textpipe/doc.py

```diff
@@ -94,59 +111,71 @@ def clean_text(self):
                 text = re.sub('[„“]|(\'\')|(,,)', '"', text)
-                text = re.sub('\s+', ' ', text)
+                text = re.sub(r'\s+', ' ', text)
                 self._clean_text = text.strip()
             else:
                 self._clean_text = ''
         return self._clean_text
 
-    @property
-    def ents(self):
+    def ents(self, model_mapping=None):
         """
         Extract a list of the named entities in text
 
         >>> doc = Doc('Sentence for testing Google text')
-        >>> doc.ents
+        >>> doc.ents()
         [('Google', 'ORG')]
         """
-        return list({(ent.text, ent.label_) for ent in self.spacy_doc.ents})
-
+        lang = self._hint_language if self.language() == 'un' else self.language()
+        model_name = model_mapping.get(lang, None) if model_mapping else None
+        try:
+            return list({(ent.text, ent.label_) for ent in self.spacy_doc(model_name).ents})
+        except TextpipeMissingModelException:
+            return list({(ent.text, ent.label_) for ent in self.spacy_doc().ents})
 
-    @property
     def nsents(self):
         """
         Extract the number of sentences from text
 
         >>> doc = Doc('Test sentence for testing text.')
-        >>> doc.nsents
+        >>> doc.nsents()
         1
         """
-        return len(list(self.spacy_doc.sents))
+        return len(list(self.spacy_doc().sents))
 
-    @property
     def nwords(self):
         """
         Extract the number of words from text
 
         >>> doc = Doc('Test sentence for testing text')
-        >>> doc.nwords
+        >>> doc.nwords()
         5
         """
-        return len(self.clean_text.split())
+        return len(self.clean_text().split())
 
-    @property
     def complexity(self):
         """
         Determine the complexity of text using the Flesch
         reading ease test ranging from 0.0 - 100.0 with 0.0
         being the most difficult to read.
 
         >>> doc = Doc('Test sentence for testing text')
-        >>> doc.complexity
+        >>> doc.complexity()
         83.32000000000004
         """
         if not self._text_stats:
-            self._text_stats = textacy.TextStats(self.spacy_doc)
+            self._text_stats = textacy.TextStats(self.spacy_doc())
         if self._text_stats.n_syllables == 0:
             return 100
         return self._text_stats.flesch_reading_ease
+
+    def _set_spacy_defaults(self):
+        """
+        Loads the spacy default language module for the Doc's language into the spacy_nlp object
+        """
+        lang = self._hint_language if self.language() == 'un' else self.language()
+        if lang not in self.allowed_languages:
+            raise TextpipeLanguageNotSupportedException
+        if lang not in self.spacy_nlps:
+            self.spacy_nlps[lang] = {}
+        model = spacy.load('{}_core_{}_sm'.format(lang, 'web' if lang == 'en' else 'news'))
+        self.spacy_nlps[lang][None] = model
```

Comment:
Above there is this snippet:
```
if None not in self.spacy_nlps:
    self._set_spacy_defaults()
```
Here this is set
```
self.spacy_nlps[lang][None] = model
```
Which seems to be unaligned.

Reply:
Fixed this.